lib/logstash/filters/grok/timeout_enforcer.rb

```diff
@@ -56,9 +56,9 @@ def stop!
 
   def cancel_timed_out!
     now = java.lang.System.nanoTime # save ourselves some nanotime calls
-    @threads_to_start_time.forEach do |thread, start_time|
+    @threads_to_start_time.each do |entry|
```

Comment:
Since we're only using the key portion of each `Entry`, should we send `Map#keySet()`, and iterate over that instead? Would doing so have the same effect?

~~~ diff
-    @threads_to_start_time.each do |thread, start_time|
+    @threads_to_start_time.keySet.each do |thread|
~~~